code/controllers/subsystem/metrics.dm

```diff
@@ -0,0 +1,59 @@
+/*
+
+	Ok listen up
+
+	This thing ingests data to ElasticSearch in a VERY SPECIFIC FORMAT
+	Not only is changing this a very bad idea due to elasticsearch being very finnicky with data formatting,
+	but if you edit this subsystem and its fields, you invalidate a lot of existing data
+
+	Dont touch this shit without speaking to crossed or AA07 first
+
+*/
+SUBSYSTEM_DEF(metrics)
+	name = "Metrics"
+	wait = 30 SECONDS
+	runlevels = RUNLEVEL_LOBBY | RUNLEVEL_SETUP | RUNLEVEL_GAME | RUNLEVEL_POSTGAME // ALL THE LEVELS
+	flags = SS_KEEP_TIMING // This needs to ingest every 30 IRL seconds, not ingame seconds.
+	/// The real time of day the server started. Used to calculate time drift
+	var/world_init_time = 0 // Not set in here. Set in world/New()
+
+/datum/controller/subsystem/metrics/Initialize(start_timeofday)
+	if(!CONFIG_GET(flag/elasticsearch_metrics_enabled))
+		flags |= SS_NO_FIRE // Disable firing to save CPU
+	return ..()
+
+
+/datum/controller/subsystem/metrics/fire(resumed)
+	var/datum/http_request/request = new()
+	request.prepare(RUSTG_HTTP_METHOD_POST, CONFIG_GET(string/elasticsearch_metrics_endpoint), get_metrics_json(), list(
+		"Authorization" = "ApiKey [CONFIG_GET(string/elasticsearch_metrics_apikey)]",
+		"Content-Type" = "application/json"
+	))
+	request.begin_async() // Fire and forget who gives a shit
+
+/datum/controller/subsystem/metrics/proc/get_metrics_json()
+	var/list/out = list()
+	out["@timestamp"] = SQLtime() // This is required by ElasticSearch, complete with this name. DO NOT REMOVE THIS.
```

Comment:
Already talked to Affected about this over Discord. We'll need a new proc that provides an ISO friendly timestamp before it can be properly test merged. We could either modify SQLtime to have an optional Boolean to output a properly compliant ISO timestamp, or, we could have a new proc. I don't personally care either way, one of the other maintainers might though. This would be my recommendation.

```dm
/proc/iso_timestamp(timevar)
    return time2text(timevar || world.timeofday, "YYYY-MM-DDThh:mm:ss")
```

Reply:
`get_metrics_json returned: {"@timestamp":"2021-08-21T12:10:47","cpu":1.07202, ...`

Done